cairo1-contracts/events.cairo

```diff
@@ -0,0 +1,65 @@
+#[contract]
+mod Contract {
+    use serde::Serde;
+    use array::ArrayTrait;
+
+    #[derive(Drop, Serde)]
+    struct EventStruct {
+        type_felt252: felt252,
+        type_u8: u8,
+        type_u16: u16,
+        type_u32: u32,
+        type_u64: u64,
+        type_u128: u128,
+        type_u256: u256,
+        type_tuple: (felt252, u8)
+    }
+
+    struct Storage {
+        balance: felt252, 
+    }
+
+
+    #[event]
+    fn BalanceChanged(prev_balance: felt252, balance: felt252) {}
+
+    #[event]
+    fn ComplexEvent(simple: felt252, event_struct: EventStruct, type_tuple: (felt252, u8, usize)) {}
+
+    #[constructor]
+    fn constructor(initial_balance: felt252) {
+        balance::write(initial_balance);
+    }
+
+    // to force EventStruct type generation in abi
+    #[external]
+    fn useless(event_struct: EventStruct) {}
```

Comment:
If this is going to be the recommended way, it needs to be documented. Has Starkware been informed about this? Is this fixed in cairo 2.0.0 or a newer version? Are event specs perhaps stored in some other part of the JSON compilation artifact, and not in the ABI?

Reply:
i created an issue about this : https://github.com/starkware-libs/cairo/issues/3419
from my quick tests it was still an issue in 2.0.0-rc2

Reply:
Right, I forgot about the issue. Can you put the link as a comment between L34 and L35? And I'll ping Starkware about it.

Reply:
done